api/agent_test.go

```diff
@@ -152,6 +153,50 @@ func TestAgent_ForceLeave(t *testing.T) {
 		wait.Timeout(3*time.Second),
 		wait.Gap(100*time.Millisecond),
 	))
+
+}
+
+func TestAgent_ForceLeavePrune(t *testing.T) {
+	testutil.Parallel(t)
+
+	c, s := makeClient(t, nil, nil)
+	defer s.Stop()
+	a := c.Agent()
+
+	nodeName := "foo.global"
+	_, s2 := makeClient(t, nil, func(c *testutil.TestServerConfig) {
+		c.NodeName = nodeName
+		c.Server.BootstrapExpect = 0
+	})
+	defer s2.Stop()
+
+	n, err := a.Join(s2.SerfAddr)
+	must.NoError(t, err)
+	must.One(t, n)
+
+	err = a.ForceLeave(nodeName, true)
+	must.NoError(t, err)
+
+	time.Sleep(3 * time.Second)
```

Comment:
This is going to be flaky. We're stopping the server and then waiting some arbitrary amount of time before seeing if the node has been removed from the member list. You can get this scenario:

* `ForceLeave`
* start waiting 3 seconds
* node leaves
* node rejoins
* 3 seconds expires
* run the `must.Wait` for 3 seconds

Even if you removed the `time.Sleep` here, the node could be removed and rejoined in the 100ms window of the `must.Wait`. So we should also be stopping the node we're removing, so that we can be sure it's not going to rejoin. And then we can safely remove this `time.Sleep`, which doesn't get us anything when we have `must.Wait` below.

Reply:
Could you help me with the test? I have updated the test following your instructions, but it is still fail. I am not sure if it has to do with how the Test Server is created.